app/src/main/java/mozilla/lockbox/store/DataStore.kt

```diff
@@ -177,11 +178,39 @@ open class DataStore(
         credentials.apply {
             support.syncConfig = SyncUnlockInfo(kid, accessToken, syncKey, tokenServerURL)
         }
+
+        if (!credentials.isNew) {
+            val state = if (backend.isLocked()) {
+                State.Locked
+            } else {
+                unlock()
```

Comment:
This can be removed if the `.Locking` state is going away.

Reply:
yeah I don't love the doubling-up on lock state management between this method and the `lock` / `unlock` methods

Reply:
:+1: